### DISCUSSION TITLE
Plot zoom on scroll?

### DISCUSSION THREAD

**Discussion by ntBre** [emilk/egui#3101] [Q&A]
I found https://github.com/emilk/egui/issues/1164, which has some discussion of this in the comments, but I'm trying to add a context menu on right-click for a plot. This conflicts with the boxed_zoom on Plot, so I was hoping to move zooming to the scroll wheel alone instead of ctrl+scroll to free right click for the menu.

I'm also open to other suggestions, such as differentiating between clicking and dragging on right click. I've tried something like
```rust
let d = response.drag_delta();
let d = d.dot(d);
if d.abs() < 1.0 {
    response.context_menu(|ui| {
	if ui.button("hello world").clicked() {
	    eprintln!("hello!");
	}
    });
}
```

to show the menu only when not dragged, but it still pops up on any pause in the dragging.

I also tried intercepting scroll events and turning them into Zoom events at the top of my MyApp::update method, but I think I must be fundamentally confused in general because none of these attempts seem to do what I want.
```rust
ctx.input_mut(|i| {
    for event in &mut i.events {
	if let Event::Scroll(v) = event {
	    if v.y.is_sign_positive() {
		*event = Event::Zoom(1.28);
	    } else {
		*event = Event::Zoom(0.78);
	    }
	}
    }
});
```

**Comment by YgorSouza** [comment]
I have PR #2714 open for that. There are a couple of methods added to `PlotUi` and a new example showing how to use them. Feedback is welcome.
